### DISCUSSION TITLE
Glicol as midi sequencer

### DISCUSSION THREAD

**Discussion by delaudio** [chaosprint/glicol#99] [Ideas]
Hi! Glicol is absolutely wonderful.
It's definitely great using it without having to install or configure multiple pieces of software.
I'm a long time user of Tidalcycles and I like using it mostly just for the midi part to sequence external instruments.
It would be nice to use Glicol to sequence external instruments too.
I know it's an audio live coding environment but It would be great to use it as midi sequencer too.
What do you think?
Thank you,

Federico

**Comment by chaosprint** [comment]
Hi Federico,

Thanks for posting this. MIDI and OSC are already on the todo list of Glicol:
https://github.com/chaosprint/glicol/issues/35

However, as you also mentioned, the main idea of Glicol is the audio rate communication between nodes, it would be better to put the MIDI logic into the JavaScript side.

On Strudel (the web porting of Tidal) web app, you can already use WebMIDI there:
https://strudel.tidalcycles.org

So for Glicol, the goal is to find a better merging between the sequencing power of Tidal and the audio advance of Glicol:
https://glicol-tidal.netlify.app/

**Reply by delaudio** [reply]
Yes, I've played with strudel and Ableton only for midi messages and it's great compared to tidalcycles + supercollider + virtual midi interface but what I like a lot from Glicol it's that can be used as a VSTi too, so it would be amazing to live code directly in Ableton Live a midi sequence and not only audio stuff. 
There are other projects like Gibber that allow to do that, but I potentially see Glicol as the perfect tool for "write once, run everywhere" :)

**Reply by chaosprint** [reply]
> Yes, I've played with strudel and Ableton only for midi messages and it's great compared to tidalcycles + supercollider + virtual midi interface but what I like a lot from Glicol it's that can be used as a VSTi too, so it would be amazing to live code directly in Ableton Live a midi sequence and not only audio stuff. There are other projects like Gibber that allow to do that, but I potentially see Glicol as the perfect tool for "write once, run everywhere" :)

Good point! I will try to add that next month.

**Reply by delaudio** [reply]
Thank you! I'll start learning rust. Hope to contribute soon :)
